File: 后端/web/src/api/attachmentCategory.js
import service from '@/utils/request'
// 分类列表
export const getCategoryList = () => {
    return service({
        url: '/attachment-categories/getCategoryList',
        method: 'get',
    })
}

// 添加/编辑分类
export const addCategory = (data) => {
    return service({
        url: '/attachment-categories/addCategory',
        method: 'post',
        data
    })
}

// 删除分类
export const deleteCategory = (data) => {
    return service({
        url: '/attachment-categories/deleteCategory',
        method: 'post',
        data
    })
}

File: 后端/server/router/admin/attachment_category.go
package admin

import (
	"heyu/server/middleware"
	"github.com/gin-gonic/gin"
)

type AttachmentCategoryRouter struct{}

func (a *AttachmentCategoryRouter) InitAttachmentCategoryRouter(Router *gin.RouterGroup) {
	attachmentCategoryRouter := Router.Group("attachment-categories").Use(middleware.OperationRecord())
	attachmentCategoryRouterWithoutRecord := Router.Group("attachment-categories")
	{
		attachmentCategoryRouter.POST("addCategory", attachmentCategoryApi.AddCategory)
		attachmentCategoryRouter.POST("deleteCategory", attachmentCategoryApi.DeleteCategory)
	}
	{
		attachmentCategoryRouterWithoutRecord.GET("getCategoryList", attachmentCategoryApi.GetCategoryList)
	}
}
